### PR TITLE
feat: Add InvokeRequest interface to replace invoke parameters

### PR DESCRIPTION
**Pull request by Alaple**
## What did you implement:

Implemented InvokeRequest interface in order to have all the parameters needed for the invoke function in one object

## How did you implement it:

Added an interface to the CloudService called InvokeRequest composed by all the parameters needed by the invoke call and changed the invoke signature to receive an InvokeRequest object.

### Core: 

__Unit tests working as expected__
image

## Todos:

_**Note: Run `npm run test-ci` to run all validation checks on proposed changes**_

- [x] Write tests and confirm existing functionality is not broken.
       **Validate via `npm test`**
- [ ] Write documentation
- [x] Ensure there are no lint errors.
       **Validate via `npm run lint-updated`**
       _Note: Some reported issues can be automatically fixed by running `npm run lint:fix`_
- [x] Ensure introduced changes match Prettier formatting.
       **Validate via `npm run prettier-check-updated`**
       _Note: All reported issues can be automatically fixed by running `npm run prettify-updated`_
- [x] Make sure code coverage hasn't dropped
- [ ] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

**_Is this ready for review?:_** YES
**_Is it a breaking change?:_** NO

### CODE REVIEW

**Comment by lgaspari** [comment]
@Alaple - You must split these changes in 3 PRs (core, aws & azure)

File: azure/src/services/azureFunctionCloudService.test.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import {
   ContainerResolver,
   CloudContainer,
   CloudService,
+  InvokeRequest,
```
Comment (lgaspari) [inline azure/src/services/azureFunctionCloudService.test.ts]: Azure changes must be in another PR, leave this PR just for `core`

Comment (Alaple) [reply azure/src/services/azureFunctionCloudService.test.ts]: Thanks for this feedback, now it only have the core.

File: core/src/services/cloudService.ts [hunk 2]
```diff
@@ -1,4 +1,25 @@
-import { StringParams } from "../common/stringParams";
+import {  StringParams } from "../common/stringParams";
```
Comment (lgaspari) [inline core/src/services/cloudService.ts]: nit: extra space after the curly brace

File: core/src/services/cloudService.ts [hunk 3]
```diff
@@ -1,4 +1,25 @@
...
+ * Implemented in order to have all the parameters needed for the invoke function in one object
+ */
+export interface InvokeRequest {
+  /**
+   * @param name Function name to invoke
+   * @param fireAndForget Wait for response if false (default behavior)
+   * @param payload HTTP request body
+   * @param headers Context headers
+   * @param pathParams StringParams with values for URL
+   * @param extraParams Extra paramters to invoke function
```
Comment (lgaspari) [inline core/src/services/cloudService.ts]: nit: Extra parameters

Comment (Alaple) [reply core/src/services/cloudService.ts]: Thanks for this feedback, It will be apply in my next commit

File: core/src/services/cloudService.ts (hunk 2 above)
Comment (Alaple) [reply core/src/services/cloudService.ts]: Thanks for this feedback, It will be apply in my next commit

File: core/src/services/cloudService.ts [hunk 4]
```diff
@@ -13,9 +34,7 @@ export interface CloudServiceOptions {
 export interface CloudService {
   /**
    * Invoke a deployed Serverless function
-   * @param name Name of function to forget
-   * @param fireAndForget Don't listen for response if true
-   * @param payload Payload to send in invocation
+   * @param params parameters needed to call the invoke
    */
-  invoke<T>(name: string, fireAndForget: boolean, payload?: any, headers?: StringParams): Promise<T>;
+  invoke<T>(params: InvokeRequest): Promise<T>;
```
Comment (mydiemho) [inline core/src/services/cloudService.ts]: I would call this `invokeOptions` instead of `params`

Comment (Alaple) [reply core/src/services/cloudService.ts]: Hi My! Thanks for this feedback, I will implement it in my next commit.

Comment (Alaple) [reply core/src/services/cloudService.ts]: Hi My! Thanks for this feedback, it will be applied in my next commit.
